pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -0,0 +1,252 @@
+        transform can be applied.
+
+        Skipping diagonalization for a circuit containing parametric mid-circuit measurements may result
+        in a completed execution with incorrect results.
+
+    Args:
+        wires (.Wires): The wires the measurement process applies to.
+            This can only be specified if an observable was not provided.
+
+    Keyword Args:
+        angle (float): The angle in radians
+        plane (str): The plane the measurement basis lies in. Options are "XY", "ZX" and "ZY"
+        reset (bool): Whether to reset the wire after measurement.
+        postselect (Optional[int]): Which basis state to postselect after a mid-circuit
+            measurement. None by default. If postselection is requested, only the post-measurement
+            state that is used for postselection will be considered in the remaining circuit.
+        id (str): Custom label given to a measurement instance.
+    """
+
+    _shortname = "measure"
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        wires: Optional[Wires],
+        *,
+        angle: Optional[float],
+        plane: Optional[str],
+        reset: Optional[bool] = False,
+        postselect: Optional[int] = None,
+        id: Optional[str] = None,
+    ):
+        self.batch_size = None
+        super().__init__(wires=Wires(wires), reset=reset, postselect=postselect, id=id)
+        self.plane = plane
+        self.angle = angle
+
+    def _flatten(self):
+        metadata = (
+            ("angle", self.angle),
+            ("wires", self.raw_wires),
+            ("plane", self.plane),
+            ("reset", self.reset),
+            ("id", self.id),
+        )
+        return (None, None), metadata
+
+    # pylint: disable=arguments-renamed, arguments-differ
+    @property
+    def hash(self):
+        """int: Returns an integer hash uniquely representing the measurement process"""
+        fingerprint = (
+            self.__class__.__name__,
+            self.plane,
+            self.angle,
+            tuple(self.wires.tolist()),
+            self.id,
+        )
+
+        return hash(fingerprint)
+
+    def __repr__(self):
+        """Representation of this class."""
+        return f"{self._shortname}_{self.plane.lower()}(wires={self.wires.tolist()}, angle={self.angle})"
+
+    @property
+    def has_diagonalizing_gates(self):
+        """Whether there are gates that need to be applied for to diagonalize the measurement"""
+        return True
+
+    def diagonalizing_gates(self):
+        """Decompose to a diagonalizing gate and a standard MCM in the computational basis"""
+        if self.plane == "XY":
+            mat = np.array(
+                [[1, np.exp(-1j * self.angle)], [1, -np.exp(-1j * self.angle)]]
+            ) / np.sqrt(2)
+            return [qml.QubitUnitary(mat, wires=self.wires)]
```

Comment:
I think (pending some checking) this could be replaced with `[H, PhaseShift(-angle)]` if we want to avoid the qubit-unitary altogether.

Reply:
Updated to Hadamard/PhaseShift diagonalization, and the tests for diagonalization results are happy!